### PR TITLE
Shortened and refactored batch methods

### PR DESCRIPTION
**Pull request by mingazhev**
1. All batch methods moved to one controller - InternalBookingsController
2. All batch methods moved to new service - BookingJobsService
3. Methods renamed accordingly to new meaning

Most of logic leaved "as is". Logic will be fixed in the next PR

### CODE REVIEW

File: Api/Services/Accommodations/Bookings/IBookingJobsService.cs [hunk 1]
```diff
@@ -7,8 +7,16 @@
 
 namespace HappyTravel.Edo.Api.Services.Accommodations.Bookings
 {
-    public interface IBookingsProcessingService
+    public interface IBookingJobsService
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/IBookingJobsService.cs:10]: The old one is OK. Job is an external term

File: Api/Services/Accommodations/Bookings/IBookingJobsService.cs [hunk 2]
```diff
@@ -7,8 +7,16 @@
...
     {
+        Task<Result<List<int>>> GetForCapture(DateTime deadlineDate);
+
+        Task<Result<ProcessResult>> Capture(List<int> bookingIds, ServiceAccount serviceAccount);
+        
+        Task<Result<List<int>>> GetForNotify(DateTime deadlineDate);
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/IBookingJobsService.cs:16]: Notify is a verb

File: Api/Controllers/InternalBookingsController.cs [hunk 3]
```diff
@@ -48,11 +49,67 @@ public async Task<IActionResult> GetBookingsForCancellation(DateTime deadlineDat
...
+            return OkOrBadRequest(await _bookingJobsService.Capture(bookingIds, serviceAccount));
+        }
+
+
+        /// <summary>
+        ///     Sends need payment notifications for bookings
+        /// </summary>
+        /// <param name="deadlineDate">Deadline date</param>
+        /// <returns>Result message</returns>
+        [HttpPost("notify/need-payment/{deadlineDate}")]
```
Comment (kirillta) [inline Api/Controllers/InternalBookingsController.cs:89]: Do you have any alternative naming options?

File: Api/Services/Accommodations/Bookings/BookingJobsService.cs [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            }
+        }
+
+
+        public async Task<Result<List<int>>> GetForCancellation(DateTime deadlineDate)
+        {
+            if (deadlineDate == default)
+                return Result.Failure<List<int>>("Deadline date should be specified");
+
+            // It’s prohibited to cancel booking after check-in date
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/BookingJobsService.cs:115]: Technically agent can cancel after a check-in date from the UI, we have a trello task for that

File: Api/Services/Accommodations/Bookings/BookingJobsService.cs [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        {
+            BookingStatusCodes.Pending, BookingStatusCodes.Confirmed
+        };
+
+        private static readonly HashSet<BookingStatusCodes> BookingStatusesForCancellation = new HashSet<BookingStatusCodes>
+        {
+            BookingStatusCodes.Pending, BookingStatusCodes.Confirmed
+        };
+
+
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/BookingJobsService.cs:217]: An extra space

File: Api/Services/Accommodations/Bookings/BookingJobsService.cs [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        private static readonly HashSet<BookingPaymentStatuses> PaymentStatusesForCancellation = new HashSet<BookingPaymentStatuses>
+        {
+            BookingPaymentStatuses.NotPaid, BookingPaymentStatuses.Authorized
+        };
+
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/BookingJobsService.cs:222]: A missing space

File: Api/Controllers/InternalBookingsController.cs (hunk 3 above)
Comment (mingazhev) [reply Api/Controllers/InternalBookingsController.cs:89]: notify/deadline-approach
deadline-notify

File: Api/Services/Accommodations/Bookings/IBookingJobsService.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/IBookingJobsService.cs:10]: Fixed

File: Api/Services/Accommodations/Bookings/IBookingJobsService.cs (hunk 2 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/IBookingJobsService.cs:16]: Fixed

File: Api/Services/Accommodations/Bookings/BookingJobsService.cs (hunk 6 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/BookingJobsService.cs:222]: Fixed

File: Api/Services/Accommodations/Bookings/BookingJobsService.cs (hunk 5 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/BookingJobsService.cs:217]: Fixed

File: Api/Services/Accommodations/Bookings/BookingJobsService.cs (hunk 4 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/BookingJobsService.cs:115]: I'll change the logic in the next PR. There I've leaved code as it was before

File: Api/Controllers/InternalBookingsController.cs (hunk 3 above)
Comment (kirillta) [reply Api/Controllers/InternalBookingsController.cs:89]: Me prefer `notify/deadline-approach`

Comment (mingazhev) [reply Api/Controllers/InternalBookingsController.cs:89]: Done

**Pull request merged**